modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -373,35 +373,20 @@ object SeqexecEngine {
       case _                 => configStatus(step.executions)
     }
 
-  protected[server] def observeStatus(executions: List[List[engine.Action]],
-                                      configStatus: List[(Resource, ActionStatus)]): ActionStatus = {
-    def containsObserve(e: List[engine.Action]): Boolean =
-      separateActions(e)._2.map(_.kind).contains(ActionType.Observe)
-
-    if (configStatus.forall(_._2 === ActionStatus.Completed)) {
-      // Find one with kind observe
-      executions.filter { e =>
-        val (a, r) = separateActions(e).bimap(_.map(_.kind), _.map(_.kind))
-        a.contains(ActionType.Observe) || r.contains(ActionType.Observe)
-      }.map {
-        case e if containsObserve(e) => ActionStatus.Completed
-        case _                       => ActionStatus.Running
-      }.headOption.getOrElse(ActionStatus.Pending)
-    } else ActionStatus.Pending
-  }
+  protected[server] def observeStatus(executions: List[List[engine.Action]]): ActionStatus =
```

Comment:
The previous change (that added an explicit state to Action) really simplified this.